lit_nlp/examples/models/dalle.py

```diff
@@ -0,0 +1,164 @@
+import transformers
+import flax
+import flax.training.common_utils
+import tqdm.notebook
+
+
+DalleBart = dalle_mini.DalleBart
+DalleBartProcessor = dalle_mini.DalleBartProcessor
+VQModel = vqgan_jax.modeling_flax_vqgan.VQModel
+CLIPProcessor = transformers.CLIPProcessor
+FlaxCLIPModel = transformers.FlaxCLIPModel
+shard_prng_key = flax.training.common_utils.shard_prng_key
+trange = tqdm.notebook.trange
+
+replicate = flax.jax_utils.replicate
+JsonDict = lit_types.JsonDict
+
+class DalleModel(lit_model.Model):
+  """Image to Text Model"""
+
+ 
+
+  def __init__(self,
+              model_name:Str,
+              predictions:int):
+    super().__init__()
+
+    self.model = model_name
+    self.predictions = predictions
+    
+  ##
+  # LIT API implementation
+  def max_minibatch_size(self) -> int:
+    return 8
+
+  def predict_minibatch(self, inputs):
+    """Model prediction based on code pipeline in doc https://github.com/borisdayma/dalle-mini"""
+    # tokenize the text. -> then return prediction
+    # Load VQGAN
+    # VQGAN model
+    VQGAN_REPO = "dalle-mini/vqgan_imagenet_f16_16384"
+    VQGAN_COMMIT_ID = "e93a26e7707683d349bf5d5c41c5b0ef69b677a9"
+
+    # small model -> dalle-mini/dalle-mini/mini-1:v0  larger one:"dalle-mini/dalle-mini/mega-1-fp16:latest"
+    DALLE_MODEL = self.model
+    DALLE_COMMIT_ID = None
+    model, params = DalleBart.from_pretrained(
+        DALLE_MODEL, revision=DALLE_COMMIT_ID, dtype=jnp.float16, _do_init=False
+    )
+    vqgan, vqgan_params = VQModel.from_pretrained(
+        VQGAN_REPO, revision=VQGAN_COMMIT_ID, _do_init=False
+    )
+    
+    params = replicate(params)
+    vqgan_params = replicate(vqgan_params)
+
+    # model inference
+    @partial(jax.pmap, axis_name="batch", static_broadcasted_argnums=(3, 4, 5, 6))
+    def p_generate(
+        tokenized_prompt, key, params, top_k, top_p, temperature, condition_scale
+    ):
+        return model.generate(
+            **tokenized_prompt,
+            prng_key=key,
+            params=params,
+            top_k=top_k,
+            top_p=top_p,
+            temperature=temperature,
+            condition_scale=condition_scale,
+        )
+
+    # decode image
+    @partial(jax.pmap, axis_name="batch")
+    def p_decode(indices, params):
+        return vqgan.decode_code(indices, params=params)
+    
+    # create a random key
+    seed = random.randint(0, 2**32 - 1)
+    key = jax.random.PRNGKey(seed)
+    
+    prompts = [ex["prompt"] for ex in inputs]
+
+    processor = DalleBartProcessor.from_pretrained(DALLE_MODEL, revision=DALLE_COMMIT_ID)
+    tokenized_prompts = processor(prompts)
+    tokenized_prompt = replicate(tokenized_prompts)
+
+    # generate Images
+    # number of predictions per prompt
+    n_predictions = self.predictions
+    # We can customize generation parameters ( https://huggingface.co/blog/how-to-generate)
+    gen_top_k = None
+    gen_top_p = None
+    temperature = None
+    cond_scale = 10.0
```

Comment:
These could use better type annotations (especially the `gen_*` properties, can't tell if they're `bool`, `int`, or something else), but also seem like great candidates for `Optional` fields in the `input_spec()`. Will be important to get the validation right for these though (e.g., can you use temp with top_k, etc.).

Reply:
Can use the types from https://github.com/borisdayma/dalle-mini/blob/main/src/dalle_mini/model/modeling.py#L1589

```suggestion
    gen_top_k: Optional[int] = None
    gen_top_p: Optional[float] = None
    temperature: Optional[float] = None
    cond_scale: Optional[float] = 10.0
```